pubsub/api/Pubsub.Samples/PullMessagesAsyncWithErrorListen.cs

```diff
@@ -0,0 +1,54 @@
+﻿// Copyright 2020 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// [START pubsub_subscriber_error_listener]
+
+using Google.Cloud.PubSub.V1;
+using System;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+
+public class PullMessagesAsyncWithErrorListen
+{
+    public async Task<int> PullMessagesAsyncListenError(string projectId, string subscriptionId, bool acknowledge)
+    {
+        SubscriptionName subscriptionName = SubscriptionName.FromProjectSubscription(projectId, subscriptionId);
+        SubscriberClient subscriber = await SubscriberClient.CreateAsync(subscriptionName);
+        // SubscriberClient runs your message handle function on multiple
+        // threads to maximize throughput.
+        int messageCount = 0;
+        Task startTask = subscriber.StartAsync((PubsubMessage message, CancellationToken cancel) =>
+        {
+            string text = System.Text.Encoding.UTF8.GetString(message.Data.ToArray());
+            Console.WriteLine($"Message {message.MessageId}: {text}");
+            Interlocked.Increment(ref messageCount);
+            return Task.FromResult(acknowledge ? SubscriberClient.Reply.Ack : SubscriberClient.Reply.Nack);
+        }).ContinueWith(task =>

```

Comment:
I am not sure what we want to demonstrate as part of this sample. But corresponding Java sample says - Listen for unrecoverable failures. Rebuild a subscriber and restart subscribing when the current subscriber encounters permanent errors. We are not doing this. Can we please check and confirm what is it that we want to demonstrate here?